### PR TITLE
Add BSD 

### PR DESCRIPTION
**Pull request by j-chim**
This PR adds BSD, the Japanese-English business dialogue corpus by 
[Rikters et al., 2020](https://www.aclweb.org/anthology/D19-5204.pdf).

### CODE REVIEW

**Comment by yjernite** [comment]
Glad to have more Japanese data! Couple of comments:
- the abbreviation might confuse some people as there is also an OPUS BSD corpus, would you mind renaming it as `bsd_ja_en`?
- `flake8` is throwing some errors, you can run it locally (`flake8 datasets`) and fix what it tells you until it's happy :)
- We're not using `os.path.join` for URLs as it's unstable across systems (introduces backslashes on Windows). Can you write the URLs explicitly instead?

Thanks!

**Comment by yjernite** [comment]
Fantastic, looks great!

**Comment by j-chim** [comment]
> Fantastic, looks great!

Thanks for your help @yjernite, really appreciate it!

File: datasets/bsd_ja_en/bsd_ja_en.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+_REPO = "https://raw.githubusercontent.com/tsuruoka-lab/BSD/master/"
+
+_URLs = {
+    "train": _REPO + "train.json",
+    "dev": _REPO + "dev.json",
+    "test": _REPO + "test.json",
+}
+
+
+class Bsd(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/bsd_ja_en/bsd_ja_en.py:81]:
```suggestion
class BsdJaEn(datasets.GeneratorBasedBuilder):
```
this must be the camel vase version of the dataset script name `bsd_ja_en`

File: datasets/bsd_ja_en/dataset_infos.json [hunk 2]
```diff
@@ -0,0 +1 @@
...type": "string", "id": null, "_type": "Value"}}, "post_processed": null, "supervised_keys": null, "builder_name": "bsd", "config_name": "default", "version": {"version_str": "1.0.0", "description": null, "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 4778409, "num_examples": 20000, "dataset_name": "bsd"}, "test": {"name": "test", "num_bytes": 493038, "num_examples": 2120, "dataset_name": "bsd"}, "validation": {"name": "validation", "num_bytes": 477964, "num_examples": 2051, "dataset_name": "bsd"}}, "download_checksums": {"https://raw.githubusercontent.com/tsuruoka-lab/BSD/master/train.json": {"num_bytes": 6740756, "checksum": "e011d1b02ed1acdbed5b677b54a3ceae26baaf6cd2a4fea64e4af4ff699c5ffb"}, "https://raw.githubusercontent.com/tsuruoka-lab/BSD/master/dev.json": {"num_bytes": 687409, "checksum": "43c11e7d9bdb4ab8ecc83114c82a77c596f98c8845af92fb74c8ca12cc9cfa5c"}, "https://raw.githubusercontent.com/tsuruoka-lab/BSD/master/test.json": {"num_bytes": 706880, "checksum": "9da3f8907147b2424671058c93d9f41179ddbd8d0c8298a3e8546c2703174d31"}}, "download_size": 8135045, "post_processing_size": null, "dataset_size": 5749411, "size_in_bytes": 13884456}}
```
Comment (lhoestq) [inline datasets/bsd_ja_en/dataset_infos.json:1]:
```suggestion
{"default": {"description": "This is the Business Scene Dialogue (BSD) dataset,\na Japanese-English parallel corpus containing written conversations\nin various business scenarios.\n\nThe dataset was constructed in 3 steps:\n  1) selecting business scenes,\n  2) writing monolingual conversation scenarios according to the selected scenes, and\n  3) translating the scenarios into the other language.\n\nHalf of the monolingual scenarios were written in Japanese\nand the other half were written in English.\n\nFields:\n- id: dialogue identifier\n- no: sentence pair number within a dialogue\n- en_speaker: speaker name in English\n- ja_speaker: speaker name in Japanese\n- en_sentence: sentence in English\n- ja_sentence: sentence in Japanese\n- original_language: language in which monolingual scenario was written\n- tag: scenario\n- title: scenario title\n", "citation": "@inproceedings{rikters-etal-2019-designing,\n    title = \"Designing the Business Conversation Corpus\",\n    author = \"Rikters, Mat\u012bss  and\n      Ri, Ryokan  and\n      Li, Tong  and\n      Nakazawa, Toshiaki\",\n    booktitle = \"Proceedings of the 6th Workshop on Asian Translation\",\n    month = nov,\n    year = \"2019\",\n    address = \"Hong Kong, China\",\n    publisher = \"Association for Computational Linguistics\",\n    url = \"https://www.aclweb.org/anthology/D19-5204\",\n    doi = \"10.18653/v1/D19-5204\",\n    pages = \"54--61\"\n}\n", "homepage": "https://github.com/tsuruoka-lab/BSD", "license": "CC BY-NC-SA 4.0", "features": {"id": {"dtype": "string", "id": null, "_type": "Value"}, "tag": {"dtype": "string", "id": null, "_type": "Value"}, "title": {"dtype": "string", "id": null, "_type": "Value"}, "original_language": {"dtype": "string", "id": null, "_type": "Value"}, "no": {"dtype": "int32", "id": null, "_type": "Value"}, "en_speaker": {"dtype": "string", "id": null, "_type": "Value"}, "ja_speaker": {"dtype": "string", "id": null, "_type": "Value"}, "en_sentence": {"dtype": "string", "id": null, "_type": "Value"}, "ja_sentence": {"dtype": "string", "id": null, "_type": "Value"}}, "post_processed": null, "supervised_keys": null, "builder_name": "bsd_ja_en", "config_name": "default", "version": {"version_str": "1.0.0", "description": null, "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 4778409, "num_examples": 20000, "dataset_name": "bsd_ja_en"}, "test": {"name": "test", "num_bytes": 493038, "num_examples": 2120, "dataset_name": "bsd_ja_en"}, "validation": {"name": "validation", "num_bytes": 477964, "num_examples": 2051, "dataset_name": "bsd_ja_en"}}, "download_checksums": {"https://raw.githubusercontent.com/tsuruoka-lab/BSD/master/train.json": {"num_bytes": 6740756, "checksum": "e011d1b02ed1acdbed5b677b54a3ceae26baaf6cd2a4fea64e4af4ff699c5ffb"}, "https://raw.githubusercontent.com/tsuruoka-lab/BSD/master/dev.json": {"num_bytes": 687409, "checksum": "43c11e7d9bdb4ab8ecc83114c82a77c596f98c8845af92fb74c8ca12cc9cfa5c"}, "https://raw.githubusercontent.com/tsuruoka-lab/BSD/master/test.json": {"num_bytes": 706880, "checksum": "9da3f8907147b2424671058c93d9f41179ddbd8d0c8298a3e8546c2703174d31"}}, "download_size": 8135045, "post_processing_size": null, "dataset_size": 5749411, "size_in_bytes": 13884456}}
```

`bsd` -> `bsd_ja_en`

**Review by lhoestq — APPROVED** [review]
Awesome ! thanks for adding this one :)

**Comment by lhoestq** [comment]
The RemoteDatasetTest is fixed on master so it's fine

**Comment by lhoestq** [comment]
merging since the CI is fixed on master

**Pull request merged**